### ISSUE TITLE
French model released

### ISSUE THREAD

**Issue by RASPIAUDIO** [SWivid/F5-TTS#434]
### Checks

- [X] This template is only for question, not feature requests or bug reports.
- [X] I have thoroughly reviewed the project documentation and read the related paper(s).
- [X] I have searched for existing issues, including closed ones, no similar questions.
- [X] I confirm that I am using English to submit this report in order to facilitate communication.

### Question details

Hi Please find an improvised tutorial on how to train a new language for F5-TTS simply using the gradio web interface

https://www.youtube.com/watch?v=UO4usaOojys

Find also the link for the french model I have trained on 80k samples, 100 epochs, on single speaker, the google drive link is in the video comment. As well as training material used in the video description. Please subscribe to my channel to support!

My feeling is that the more I train the less sensitive it gets to the reference sample that I want to clone and that the result come closer to the training sample, any idea how to overcome this?

**Comment by SWivid** [comment]
Hi @RASPIAUDIO , will need diverse training corpus. It's normal case if only single speaker to train.

**Comment by x4080** [comment]
@RASPIAUDIO I watch your video and its very informative, do you know if the chunks to train have any maximum limitation of length ? ex max is only 15 secs ? Or it can handle unlimited length

**Comment by SWivid** [comment]
no limit if you have unlimited gpu mem, just change the `model/dataset.py` remove corresponding duration constraints

**Comment by RASPIAUDIO** [comment]
Thanks I didin't know that it could be changed easily. On a single 4090 I'm using default settings, it make usually sample of 5s sometime up to 7s rarely 10s. I will try to increase sample lenght to see the influence on the training quality.

**Comment by x4080** [comment]
@SWivid @RASPIAUDIO Thanks for answering

**Comment by x4080** [comment]
I have another question, if the training data include numbers, should we convert the number into text or let it be number and the model will take care of it by itself ?

**Comment by SWivid** [comment]
if single language/no confict of using, it's fine (e.g. currently numbers will be read in English), otherwise a convertion is needed.

**Comment by x4080** [comment]
@SWivid Thanks, its in another language, How can I continue the training, will saving the last .pt enough ? I'm currently using rented GPU to train and I evaluate the result then wanted to continue training in another day

**Comment by SWivid** [comment]
> will saving the last .pt enough

yes, if the last one is the best after evaluation (say, no overfitting)

> Edit : What is the target loss is it under 0.5 ?

no idea about it (cuz different dataset

**Comment by SWivid** [comment]
Thanks @RASPIAUDIO , if you are willing, could also make PR to https://github.com/SWivid/F5-TTS/blob/main/src/f5_tts/infer/SHARED.md

**Comment by x4080** [comment]
@SWivid Thanks, the problem that I had now is that it seems that the training stuck at the last epoch, it just staying there while cpu and gpu is at full throttle, I tried it for example in 5 epoch, but if I use 5000 epoch it passed the 5th epoch

**Comment by x4080** [comment]
@RASPIAUDIO after round 2 did the model support zero shot correctly ? Like following the sample speaker that it never hears ?

**Comment by SWivid** [comment]
will close this issue, feel free to open if further questions

**Comment by SmaugPool** [comment]
Thank you for this great French model @RASPIAUDIO!

If I want to train a single French speaker for maximum similarity (inference not good enough), do you think it's better to train from scratch or from the final French multi-speaker model?

**Comment by RASPIAUDIO** [comment]
> Thank you for this great French model @RASPIAUDIO!
> 
> If I want to train a single French speaker for maximum similarity (inference not good enough), do you think it's better to train from scratch or to finetune from the final French multi-speaker model?

Finetune

**Comment by ayetee-re** [comment]
I like your tutorial, thank you for sharing. Is there any chance you could extend it showing how to finetune pretrained model? From what I see you've made 2 runs so I suppose you've done this already.

**Comment by VinceDu66** [comment]
![tmp30cpj8rs](https://github.com/user-attachments/assets/cb68c33d-7bef-4d88-a1f0-300d8329b24a)
[audio.webm](https://github.com/user-attachments/assets/64c06625-09f3-4a72-9030-b9acf94d48ca)

**Comment by romain130492** [comment]
In order to keep the emotions as the original model does, would you train/fine tune that french model on more diversified content?

**Comment by laurentbenaroya** [comment]
Hello,
Thank you very much for this great work! 
Could you tell more precisely what data you used for training? I want to build the same model for BigVGAN (which essentially means changing the way mel spectrograms are calculated).
